src/compute/src/render/join/mod.rs

```diff
@@ -13,3 +13,5 @@
 
 mod delta_join;
 mod linear_join;
+
+pub use linear_join::JoinImpl as LinearJoinImpl;
```

Comment:
Why not just name `JoinImpl` `LinearJoinImpl`? This would remove one level of indirection.